apis/Google.Cloud.Storage.V1/Google.Cloud.Storage.V1/UrlSigner.PostPolicy.cs

```diff
@@ -0,0 +1,315 @@
+            /// </summary>
+            SuccessActionStatus = 10,
+        }
+
+        /// <summary>
+        /// A custom element that can be added to a <see cref="PostPolicy"/>.
+        /// </summary>
+        public enum PostPolicyCustomElement
+        {
+            /// <summary>
+            /// Custom element used for requiering extra Google metadata to be included
+            /// in the post request. If the element name you provide is not prefixed by
+            /// x-goog-meta-, it will be prefixed with x-goog-meta- before being added to
+            /// the <see cref="PostPolicy"/>.
+            /// </summary>
+            GoogleMetadata = 0,
+
+            /// <summary>
+            /// Custom element used for requiering extra fields to be included in the post request.
+            /// </summary>
+            Custom = 1,
+        }
+
+        /// <summary>
+        /// Represents an unsigned post policy. It needs to be signed with <see cref="UrlSigner.Sign(PostPolicy, Options)"/> 
+        /// before it is used to enforced policies when posting objects via form.
+        /// </summary>
+        public sealed class PostPolicy
+        {
+            private static readonly IPostPolicyElement[] s_knownElements = new IPostPolicyElement[]
+            {
+                Acl.Element,
+                BucketName.Element,
+                CacheControl.Element,
+                ContentDisposition.Element,
+                ContentEncoding.Element,
+                ContentLength.Element,
+                ContentType.Element,
+                ObjectExpiry.Element,
+                Key.Element,
+                SuccessActionRedirect.Element,
+                SuccessActionStatus.Element
+            };
+
+            private static readonly ICustomElementFactory[] s_customElementFactories = new ICustomElementFactory[]
+            {
+                GoogleMetadataElementFactory.Default,
+                CustomElementFactory.Default
+            };
+
+            private readonly IDictionary<IPostPolicyElement, IPostPolicyCondition> _conditions =
+                new Dictionary<IPostPolicyElement, IPostPolicyCondition>();
+
+            private PostPolicy() { }
+
+            internal PostPolicy(PostPolicy other)
+            {
+                GaxPreconditions.CheckNotNull(other, nameof(other));
+                foreach (var condition in other._conditions)
+                {
+                    _conditions.Add(condition);
+                }
+            }
+
+            /// <summary>
+            /// Creates a new <see cref="PostPolicy"/> instance which requieres
+            /// an exact match for the <see cref="Key"/> element.
+            /// </summary>
+            /// <param name="exactKeyValue">The value to match against. Must not be null.</param>
+            /// <returns>A new <see cref="PostPolicy"/> instance.</returns>
+            public static PostPolicy FromExactKey(string exactKeyValue)
```

Comment:
This looks a bit specific in that only one field is set, but might be misunderstanding.

Reply:
Follow-up from discussion above, can it include bucket name as well?